### PR TITLE
Fix post editor dirty state caused by publicize store

### PR DESCRIPTION
**Pull request by manzoorwanijk**
Opening the Jetpack sidebar on the post edit page results in dirty state of the post editor and thus preventing users from reloading or navigating away from the page.

Screenshot 2023-11-13 at 12 39 17 PM

## Proposed changes:

* Move the connections state and actions from `jetpack/publicize` to `jetpack-social-store` and update consumer components to use the new store.
* The source of truth for the connections will now be `jetpack-social-store` instead of post meta.
* Connections in post meta will only be synced to post meta when a user explicitly toggles a connection

**Note:**
After this PR, the store inside `projects/js-packages/publicize-components/src/store` technically contains the utility functions not the actual selectors or actions because they mostly talk to the `core/editor` store not the `jetpack/publicize` store. That means we can convert those selectors/actions to simple utility functions and get rid of `jetpack/publicize` store.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Testing instructions:

- Run `jetpack build plugins/jetpack`
- Ensure that you have some social connections added for the site
- Goto post edit page for an existing post and open Jetpack sidebar
- Wait for the `connection-test-results` API request to finish
- Try to reload the page
- Confirm that their is no confirmation popup about unsaved changes
- Toggle a connection
- Confirm that the UI reflects the toggled connection state
- Try to reload the page
- Confirm that this time you see the popup
- Confirm that the social preview details like name and profile picture work correctly
- Create a new post and publish it
- Confirm that it is posted to the social media connections

### CODE REVIEW

**Comment by manzoorwanijk** [comment]
Sorry, I had to push a few more changes to reduce some duplication to get the user profile details from the connection data.

File: projects/js-packages/publicize-components/src/components/social-previews/linkedin.js [hunk 1]
```diff
@@ -13,7 +14,9 @@ import { getLinkedInDetails } from '../../store/selectors';
 export function LinkedIn( props ) {
 	const { title, url, image, media } = props;
 
-	const { name, profileImage } = getLinkedInDetails();
+	const { displayName: name, profileImage } = useSelect( select =>
+		select( SOCIAL_STORE_ID ).getConnectionProfileDetails( 'linkedin' )
```
Comment (pablinos) [inline projects/js-packages/publicize-components/src/components/social-previews/linkedin.js]: Should these connection strings be constants somewhere?

**Review by pablinos — COMMENTED** [review]
This is working well for me, nice work! Do you think it would be worth adding some more tests?

File: projects/js-packages/publicize-components/src/components/social-previews/linkedin.js (hunk 1 above)
Comment (manzoorwanijk) [reply projects/js-packages/publicize-components/src/components/social-previews/linkedin.js]: That is a good suggestion, thanks.

Comment (manzoorwanijk) [reply projects/js-packages/publicize-components/src/components/social-previews/linkedin.js]: Done ✅

File: projects/js-packages/publicize-components/src/social-store/constants.js [hunk 2]
```diff
@@ -0,0 +1,7 @@
+export const CONNECTION_SERVICE_FACEBOOK = 'facebook';
+export const CONNECTION_SERVICE_INSTAGRAM_BUSINESS = 'instagram-business';
+export const CONNECTION_SERVICE_LINKEDIN = 'linkedin';
+export const CONNECTION_SERVICE_MASTODON = 'mastodon';
+export const CONNECTION_SERVICE_NEXTDOOR = 'nextdoor';
+export const CONNECTION_SERVICE_TUMBLR = 'tumblr';
+export const CONNECTION_SERVICE_TWITTER = 'twitter';
```
Comment (pablinos) [inline projects/js-packages/publicize-components/src/social-store/constants.js]: Any particular reason for the separate file? It seems like they should coexist with the SOCIAL_STORE_ID.

Comment (manzoorwanijk) [reply projects/js-packages/publicize-components/src/social-store/constants.js]: No particular reason. I can move those to the index file.

Comment (manzoorwanijk) [reply projects/js-packages/publicize-components/src/social-store/constants.js]: Done

**Comment by manzoorwanijk** [comment]
> Do you think it would be worth adding some more tests?

I have the tests ready and I can create another PR for that to keep this one smaller.

**Review by pablinos — APPROVED** [review]
This is working well for me and cleans up the state management a lot.

**Review by spsiddarthan — APPROVED** [review]

**Pull request merged**